client/src/main/java/com/vesoft/nebula/client/storage/StorageClient.java

```diff
@@ -0,0 +1,617 @@
+     *
+     * @param spaceName nebula graph space
+     * @param edgeName  nebula edge name
+     * @param part      part to scan
+     * @param noColumns return no cols
+     * @param limit     return data limit
+     * @param startTime start time
+     * @param endTime   end time
+     * @return
+     */
+    public ScanEdgeResultIterator scanEdge(String spaceName,
+                                           int part,
+                                           String edgeName,
+                                           boolean noColumns,
+                                           int limit,
+                                           long startTime,
+                                           long endTime,
+                                           boolean allowPartSuccess) {
+        return scanEdge(spaceName, Arrays.asList(part), edgeName, new ArrayList<>(), noColumns,
+                limit, startTime, endTime, allowPartSuccess, true);
+    }
+
+
+    /**
+     * scan edge of all parts with no return cols
+     *
+     * @param spaceName nebula graph space
+     * @param edgeName  nebula edge name
+     * @param noColumns return no cols
+     * @param limit     return data limit
+     * @param startTime start time
+     * @param endTime   end time
+     * @return
+     */
+    public ScanEdgeResultIterator scanEdge(String spaceName,
+                                           String edgeName,
+                                           boolean noColumns,
+                                           int limit,
+                                           long startTime,
+                                           long endTime,
+                                           boolean allowPartSuccess,
+                                           boolean allowReadFromFollower) {
+
+        List<Integer> parts = metaManager.getSpaceParts(spaceName);
+        if (parts.isEmpty()) {
+            throw new IllegalArgumentException("No valid part in space " + spaceName);
+        }
+        return scanEdge(spaceName, parts, edgeName, new ArrayList<>(), noColumns,
+                limit, startTime, endTime, allowPartSuccess, allowReadFromFollower);
+    }
+
+
+    /**
+     * scan edge with parts
+     *
+     * @param spaceName  nebula graph space
+     * @param parts      parts to scan
+     * @param edgeName   nebula edge name
+     * @param returnCols return cols
+     * @param noColumns  if return no col, if noColumns is true, returnCols is useless
+     * @param limit      return data limit
+     * @param startTime  start time
+     * @param endTime    end time
+     * @return
+     */
+    private ScanEdgeResultIterator scanEdge(String spaceName,
+                                            List<Integer> parts,
+                                            String edgeName,
+                                            List<String> returnCols,
+                                            boolean noColumns,
```

Comment:
ditto

Reply:
This is a private method to cover both returnCols and noColumns.